### PR TITLE
Docstring cleanup and improvement of style guide compliance

### PR DESCRIPTION
**Pull request by bwohlberg**
Docstring cleanup and improvement of style guide compliance.

### CODE REVIEW

File: scico/_autograd.py [hunk 1]
```diff
@@ -69,43 +56,21 @@ def conjugated_grad(*args, **kwargs):
     return conjugated_grad_aux if has_aux else conjugated_grad
 
 
+# Append docstring from original jax function
+grad.__doc__ = _append_jax_docs(grad)
```
Comment (Michael-T-McCann) [inline scico/_autograd.py:60]: These docs aren't rendering correctly (e.g. see the text after Args: in https://scico--119.org.readthedocs.build/en/119/_autosummary/scico.html#scico.grad). I believe I got something similar working in `scico.random` after a lot of fighting, see https://github.com/lanl/scico/blob/76ad408e825c1549df4a26a2c96dd87fb95acefb/scico/random.py#L106

**Review by Michael-T-McCann — APPROVED** [review]
Looks good!

File: scico/_autograd.py (hunk 1 above)
Comment (bwohlberg) [reply scico/_autograd.py:60]: Strange, I thought it rendered correctly when I tried to build docs locally. But I guess I missed that. Thanks for the suggestion; I'll give it a try.

Comment (bwohlberg) [reply scico/_autograd.py:60]: Not so simple, unfortunately.

Comment (bwohlberg) [reply scico/_autograd.py:60]: Fixed in 376c06a.

File: scico/random.py [hunk 2]
```diff
@@ -186,21 +189,25 @@ def randn(
...
+        seed: Seed for new PRNGKey. Default: 0.
+        dtype: dtype for returned value. Default to ``np.float32``.
+            If ``np.complex64``, generates an array sampled from complex
+            normal distribution.
 
     Returns:
         tuple: A tuple (x, key) containing:
 
-           - **x** : (DeviceArray):  Generated random array
+           - **x** : (DeviceArray):  Generated random array.
```
[the remark this answers is not in the document]
Comment (bwohlberg) [reply scico/random.py:210]: To make the variables names bold, for compatibility with the parameters section.

**Pull request merged**